packages/botfuel-dialog/src/dialogs/prompt-dialog.js

```diff
@@ -177,10 +177,16 @@ class PromptDialog extends Dialog {
             newValue,
             remainingCandidates,
           });
-          const isFulfilled = dialogParameter.isFulfilled(newValue, { dialogEntities });
+          const newDialogEntities = {
+            ...matchedEntities,
+            [name]: newValue,
+          };
+          const isFulfilled = dialogParameter.isFulfilled(newValue, {
+            dialogEntities: newDialogEntities,
+          });
           return {
             // Store the found entities here as a { <entityName>: <entity> } map
-            matchedEntities: { ...matchedEntities, [name]: newValue },
+            matchedEntities: newDialogEntities,
```

Comment:
I find it a bit annoying that, in this class, matchedEntities and dialogEntities both designate entities that have been matched ... but maybe it is difficult to change